src/ui/components/RecipeEnrollmentPage.tsx

```diff
@@ -1,38 +1,145 @@
-import { ChangeEvent, FC, useState, useCallback } from "react";
-import { Form, Container, Button } from "react-bootstrap";
+import { ChangeEvent, FC, useState, useCallback, useMemo } from "react";
+import { Form, Container, Button, Modal } from "react-bootstrap";
+import { NimbusExperiment } from "@mozilla/nimbus-schemas";
 
 import { useToastsContext } from "../hooks/useToasts";
 
+type EnrollInExperimentResult =
+  | {
+      enrolled: true;
+      error: null;
+    }
+  | {
+      enrolled: false;
+      error: {
+        slugExistsInStore: boolean;
+        activeEnrollment: string | null;
+      };
+    };
+
 const RecipeEnrollmentPage: FC = () => {
   const [jsonInput, setJsonInput] = useState("");
+  const [enrollError, setEnrollError] =
+    useState<EnrollInExperimentResult["error"]>(null);
   const { addToast } = useToastsContext();
 
+  const recipe = useMemo<NimbusExperiment | null>(() => {
+    try {
+      return JSON.parse(jsonInput) as NimbusExperiment;
+    } catch {
+      return null;
+    }
+  }, [jsonInput]);
+
+  const slug = recipe?.slug || "";
+
   const handleInputChange = useCallback(
     (event: ChangeEvent<HTMLTextAreaElement>) => {
       setJsonInput(event.target.value);
     },
-    [],
+    [setJsonInput],
   );
 
-  const handleEnrollClick = useCallback(async () => {
-    try {
-      const result = await browser.experiments.nimbus.enrollInExperiment(
-        JSON.parse(jsonInput) as object,
-      );
+  const handleEnrollClick = useCallback(
+    async (
+      event?: React.MouseEvent<HTMLButtonElement>,
+      forceEnroll = false,
+    ) => {
+      event?.preventDefault();
```

Comment:
do we need to preventDefault here? What default event handler are we pre-empting?
The <Button> its attached to shouldnt trigger anything (unlike an input type=submit)